src/main/java/seedu/address/logic/commands/StatisticsCommand.java

```diff
@@ -0,0 +1,98 @@
+            + "Example: " + COMMAND_WORD;
+    public static final String MESSAGE_DISPLAY_STATISTICS_SUCCESS = "Here are all the statistics:\n%s";
+    public static final String MESSAGE_DISPLAY_TOTAL_PEOPLE = "Total Number Of People: %s";
+    public static final String MESSAGE_DISPLAY_HIGH_PRIORITY = "Number Of HIGH Priority People: %s";
+    public static final String MESSAGE_DISPLAY_MEDIUM_PRIORITY = "Number Of MEDIUM Priority People: %s";
+    public static final String MESSAGE_DISPLAY_LOW_PRIORITY = "Number Of LOW Priority People: %s";
+    private String resultMessage;
+
+    /**
+     * Displays all the overall statistics to be shown.
+     *
+     * @throws CommandException
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+        String s = "";
+        int nbOfPeople = lastShownList.size();
+        int highPriority = 0;
+        int mediumPriority = 0;
+        int lowPriority = 0;
+        String[] allStats = new String[4];
+        allStats[0] = String.format(MESSAGE_DISPLAY_TOTAL_PEOPLE, nbOfPeople);
+        allStats[1] = String.format(MESSAGE_DISPLAY_HIGH_PRIORITY, highPriority);
+        allStats[2] = String.format(MESSAGE_DISPLAY_MEDIUM_PRIORITY, mediumPriority);
+        allStats[3] = String.format(MESSAGE_DISPLAY_LOW_PRIORITY, lowPriority);
+        for (int i = 0; i < nbOfPeople; i++) {
+            Priority pri = lastShownList.get(i).getPriority();
+
+            switch (pri) {
+
+            case HIGH:
+                highPriority++;
+                break;
+            case MEDIUM:
+                mediumPriority++;
+                break;
+
+            case LOW:
+                lowPriority++;
+                break;
+
+            default:
+                throw new CommandException(Messages.MESSAGE_LACK_PRIORITY);
+            }
+        }
+        allStats[1] = String.format("Number Of HIGH Priority People: %s", highPriority);
+        allStats[2] = String.format("Number Of MEDIUM Priority People: %s", mediumPriority);
+        allStats[3] = String.format("Number Of LOW Priority People: %s", lowPriority);
+        for (int i = 0; i < allStats.length; i++) {
+            s += allStats[i] + "\n";
+        }
+        resultMessage = s;
+        return new CommandResult(String.format(MESSAGE_DISPLAY_STATISTICS_SUCCESS, resultMessage));
+    }
+
+    public String getResultMessage() {
+        return this.resultMessage;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof StatisticsCommand)) {
+            return false;
+        }
+
+        StatisticsCommand otherStatisticsCommand = (StatisticsCommand) other;
+        return resultMessage.equals(otherStatisticsCommand.getResultMessage());
```

Comment:
`resultMessage` is null when `StatisticsCommand` object is first instantiated, hence `NullPointerException` will occur when the command has not yet been executed. Can consider assigning `resultMessage` attribute with a starting value of  `""`.

Reply:
noted will do so